src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/AppInsights/WebJobsTelemetryInitializer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    internal class WebJobsTelemetryInitializer : ITelemetryInitializer
+    {
+        internal const string AzureWebsiteName = "WEBSITE_SITE_NAME";
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                return;
+            }
+
+            // RoleName is the app name
+            telemetry.Context.Cloud.RoleName = Environment.GetEnvironmentVariable(AzureWebsiteName) ?? "[Unknown]";
```

Comment:
Yeah, I complained about it already: https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/248 Not sure it was properly addressed.

If you just follow the suite - you can just enable `AzureWebAppRoleEnvironmentTelemetryInitializer` on your telemetry configuration by doing 
`config.TelemetryInitializers.Add(new AzureWebAppRoleEnvironmentTelemetryInitializer())` so you'd avoid code duplication and the implementation will be aligned with what we are doing for web apps.